### PR TITLE
Fix queries losing inherited data when toBuilder is called.

### PR DESCRIPTION
**Pull request by PiSoup**
### Description
Calling `toBuilder` on classes which implement `QueryBase` will now preserve `boost` and `queryName`.

### Issues Resolved
Fixes https://github.com/opensearch-project/opensearch-java/issues/1172

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ This section is for maintaining a changelog for all breaking changes for the cli
 
 ### Fixed
 - Fix version and build ([#254](https://github.com/opensearch-project/opensearch-java/pull/254))
+- Fix queries not preserving boost and name when converted to builders ([#1181](https://github.com/opensearch-project/opensearch-java/pull/1181))
```
Comment (PiSoup) [inline CHANGELOG.md:32]: Should I be including this in the changelog here?

Comment (dblock) [reply CHANGELOG.md:32]: Yes, thank you!

File: java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/BoolQuery.java [hunk 2]
```diff
@@ -172,7 +172,13 @@ protected void serializeInternal(JsonGenerator generator, JsonpMapper mapper) {
     }
 
     public Builder toBuilder() {
-        return new Builder().filter(filter).minimumShouldMatch(minimumShouldMatch).must(must).mustNot(mustNot).should(should);
+        return new Builder().queryName(queryName())
```
[the remark this answers is not in the document]
Comment (PiSoup) [reply java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/BoolQuery.java:175]: Yeah, that'd be nicer than calling the getters. I'll make the change.

**Review by Xtansia — APPROVED** [review]

**Review by Xtansia — CHANGES_REQUESTED** [review]
@PiSoup Changes look great thanks! Could you please just fix your DCO signoff on your two most recent commits? Should be able to do it via `git rebase --signoff HEAD~3`

**Review by reta — APPROVED** [review]

**Comment by Xtansia** [comment]
@PiSoup Unfortunately DCO is still unhappy as your commit author email is different to the email that signs off the commits:
Screenshot 2024-09-04 at 11 09 43 AM

**Comment by PiSoup** [comment]
> @PiSoup Unfortunately DCO is still unhappy as your commit author email is different to the email that signs off the commits: Screenshot 2024-09-04 at 11 09 43 AM

Oh for Pete's sake. Third time's the charm hopefully.

**Review by Xtansia — APPROVED** [review]

**Comment by Xtansia** [comment]
@PiSoup Thanks for sticking with it despite the pain 😅

**Pull request merged**